packages/zoom-based-markers/src/index.js

```diff
@@ -0,0 +1,114 @@
+import cloneDeep from "lodash.clonedeep";
+import PropTypes from "prop-types";
+import React from "react";
+
+import { zoomBasedSymbolType } from "@opentripplanner/core-utils/lib/types";
+
+/**
+ * Transforms the symbol and symbols by type from the specified symbolEntry
+ * using the specified symbolTransform.
+ * TODO: Should this be memoized?
+ */
+const getTransformedSymbol = (symbolEntry, symbolTransform) => {
+  // If no transform function provided, just return symbolEntry.
+  if (typeof symbolTransform !== "function") {
+    return symbolEntry;
+  }
+
+  const { symbolByType } = symbolEntry;
+  const newEntry = cloneDeep(symbolEntry);
+
+  if (symbolByType) {
+    Object.entries(symbolByType).forEach(([key, originalSymbol]) => {
+      newEntry.symbolByType[key] = symbolTransform(originalSymbol);
+    });
+  }
+
+  newEntry.symbol = symbolTransform(symbolEntry.symbol);
```

Comment:
Would be nice to have some comments along the way in this method that help explain what is going on. For example, above line 21, it's transforming the symbols by type and then at line 27 it transforms the default symbol.

Reply:
Addressed in https://github.com/opentripplanner/otp-ui/pull/159/commits/dcc3b3943c44cffeb34507e9127e8c3b9190098a.